modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala

```diff
@@ -208,6 +208,34 @@ final class Contexts[F[_]](agg: ContextAggregate[F], doms: Domains[F], baseUri:
       case Current(_, rev, _, value, published, deprecated) => Some(Context(id, rev, value, deprecated, published))
     }
 
+  /**
+    * Queries the system for the context identified by the argument id at the specified revision.  The (in)existence of
+    * the context or the requested revision is represented by the [[scala.Option]] type wrapped within the ''F[_]''
+    * context.
+    *
+    * @param id  the unique identifier of the context
+    * @param rev the revision attempted to be fetched
+    * @return an optional [[ch.epfl.bluebrain.nexus.kg.core.contexts.Context]] schema wrapped in the
+    *         abstract ''F[_]'' type if successful, or a [[ch.epfl.bluebrain.nexus.kg.core.Fault]] wrapped within
+    *         ''F[_]'' otherwise
+    */
+  def fetch(id: ContextId, rev: Long): F[Option[Context]] =
+    fetchCurrent(id, rev).map {
+      case None    => None
+      case Some(c) => Some(Context(c.id, rev, c.value, c.deprecated, c.published))
+    }
+
+  private def fetchCurrent(id: ContextId, rev: Long): F[Option[Current]] =
+    agg
+      .foldLeft[ContextState](id.show, Initial) {
```

Comment:
Stitch together the fold and the map to avoid the extra scheduling of the transformation at the end.

Reply:
it also makes it safer as you don't need the second pattern match in the transformation.